### DISCUSSION TITLE
Using FastAPI as the backend, I'm currently facing a CORS (Cross-Origin Resource Sharing) issue. How can I resolve it?

### DISCUSSION THREAD

**Discussion by RuoCJ** [fastapi/fastapi#12493] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py
from fastapi import FastAPI
from fastapi.routing import APIRoute
from fastapi.middleware.cors import CORSMiddleware

from app.api.main import api_router
from app.core.config import settings
from app.core.middleware import JWTMiddleware
from app.logging_config import setup_logging

app = FastAPI()

app.add_middleware(
    CORSMiddleware,  # CORS中间件
    allow_origins=[  # 允许的来源列表
        str(origin).strip("/") for origin in settings.BACKEND_CORS_ORIGINS
    ],
    allow_credentials=True,  # 是否允许发送 cookies
    allow_methods=["*"],  # 允许的 HTTP 方法
    allow_headers=["*"],  # 允许的 HTTP 头
)
app.add_middleware(JWTMiddleware)

# middleware.py
import logging
import jwt
from fastapi.responses import JSONResponse
from datetime import datetime
from starlette.middleware.base import BaseHTTPMiddleware

from app.api.deps import get_redis, get_db
from app.core.config import settings
from app.core.security import ALGORITHM
from app.schemas.response_schemas import Code, Message, RestResult
from app.utils import parse_token
from app.constant import WhiteList

class JWTMiddleware(BaseHTTPMiddleware):
    async def dispatch(self, request, call_next):
        authorization: str = request.headers.get("authorization")


# mine.py
import logging
import traceback

from fastapi import APIRouter, Request, Form
from fastapi.responses import JSONResponse
from sqlalchemy.exc import SQLAlchemyError, IntegrityError

from app.schemas.response_schemas import RestResult, Code, Message
from app.dao.user_center.mine_dao import update_user_info_dao

router = APIRouter()
@router.get(
    "/user_account",
)
async def my_account(
        request: Request
) -> JSONResponse:
    return True
```

### Description

A frontend developer is using a GET request to call the API endpoint /api/v1/user_account, and they are encountering a CORS (Cross-Origin Resource Sharing) error. I have ensured that the frontend user's origin is included in settings.BACKEND_CORS_ORIGINS. Here is a screenshot of the request:
![image](https://github.com/user-attachments/assets/bfb24afb-23a1-4b03-9b2e-654f5d5dea83)

### Operating System

Linux

### Operating System Details

linux version: Ubuntu 24.04

### FastAPI Version

0.112.1

### Pydantic Version

2.8.2

### Python Version

Python 3.10.11

**Comment by Kfir-G** [comment, marked as answer]
@RuoCJ  To fix the CORS issue in FastAPI, ensure that `settings.BACKEND_CORS_ORIGINS` is correctly set to include the frontend's origin. Double-check the exact URL (including protocols like `http://` or `https://`) in your list of allowed origins. If you're still facing the issue, try using `allow_origins=["*"]` temporarily to allow all origins and narrow down the problem. 

Example:
```python
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],  # Allow all origins temporarily
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```
Also in your screen there is a token so I recommend you to not exposed it.
